adapters/repos/db/vector/flat/index.go

```diff
@@ -261,11 +322,17 @@ func (index *flat) searchByVectorBQ(vector []float32, k int, allow helpers.Allow
 		return nil, nil, err
 	}
 
-	if err := index.findTopVectors(heap, allow, ef,
-		index.store.Bucket(helpers.VectorsFlatBQBucketLSM).Cursor,
-		index.createDistanceCalcBQ(vectorBQ),
-	); err != nil {
-		return nil, nil, err
+	if index.bqCache != nil {
```

Comment:
maybe a couple of bool methods could be added to the struct to abstract these kind of checks? like:
- `isBQ()`
- `isBQCached()`
- `isPQ()` 
- `isPQCached()`

WDYT?